angular-app/firestore.rules

```diff
@@ -1,7 +1,25 @@
+rules_version = '2';
 service cloud.firestore {
   match /databases/{database}/documents {
-    match /{document=**} {
-      allow read, write;
+    function isSignedIn() {
+    	return request.auth.uid != null;
+    }
+    
+    match /projects/{projectID} {
+    	function isOwnedBy(userId) {
+    		return get(/databases/$(database)/documents/projects/$(projectID)).data.roles[userId] == "owner";
+    	}
+    	function hasCollaborator(userId) {
```

Comment:
Would this be more appropriately named as `isCollaborator`?

Reply:
Well . . . maybe? But this naming scheme is consistent with `isOwnedBy` and refers to the document in question --- it's not an inherent property of the user. Could you maybe explain your reasoning a little more

Reply:
I see. I think I was confused, because there's an implicit [project] in front of it, i.e. `project.isOwnedBy(userId)` / `project.hasCollaborator(userId)`.

Reply:
Do you have any suggestions that would clarify the function, since I also think the implicit `project` isn't especially clear